src/AssemblyGenerator.Types.cs

```diff
@@ -155,6 +156,23 @@ private void CreateType(Type type, List<DelayedWrite> genericParams)
             CreateMethods(type.GetMethods(AllMethods), genericParams);
         }
 
+        private static string Unescape(string str)
```

Comment:
I would suggest to add an comment in the method first clarifying the intent, and second pointing the online documentation https://docs.microsoft.com/en-us/dotnet/framework/reflection-and-codedom/specifying-fully-qualified-type-names#specify-special-characters (the link may die, hence the comment). Indeed, the intent of this method can be quite cryptic otherwise.

Also, the `Unescape(string str)` method is perfect for direct unit testing. I would suggest to make it `internal`, and add a couple of more stringent unit tests.  Indeed, right now, unit test does not cover situations with multiple escape. The potential for off-by-one mistakes is high.

To make internals visible to the unit tests. https://stackoverflow.com/questions/358196/c-sharp-internal-access-modifier-when-doing-unit-testing